(this.webpackJsonptaxodium=this.webpackJsonptaxodium||[]).push([[0],{27:function(e,t,a){},32:function(e,t,a){},38:function(e,t,a){"use strict";a.r(t);var n=a(1),s=a.n(n),r=a(23),o=a.n(r),c=(a(32),a(10)),l=a(8),i=(a(27),a(2));var d=function(e){var t=e.enabled,a=e.setEnabled;return Object(i.jsxs)("div",{className:"w-full ".concat(t?"":"hidden"),children:[Object(i.jsx)("div",{onClick:function(){return a(!1)},className:"fixed w-full h-full bg-black opacity-80 z-40"}),Object(i.jsxs)("div",{onClick:function(){return a(!1)},className:"fixed w-full h-full  opacity-100 z-40",children:[Object(i.jsxs)("div",{className:" relative mx-auto mt-5 p-5 bg-white shadow-md z-100 w-4/5 overflow-y-auto opacity-100 ",style:{height:"90vh"},children:[Object(i.jsx)("button",{className:"absolute top-5 right-5 text-xl font-bold",onClick:function(){return a(!1)},children:"X"}),Object(i.jsx)("h1",{className:"font-bold mb-3 text-xl",children:"About Cov2Tree"}),"This website allows you to explore a phylogenetic tree with more than a million SARS-Cov-2 sequences, sequenced by researchers around the world. The interface was made by"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"http://theo.io/",children:"Theo Sanderson"})," ","using a custom-developed open-source library called"," ",Object(i.jsxs)("a",{href:"http://github.com/theosanderson/taxodium",className:"text-blue-700 underline",children:[" ","Taxodium"]}),".",Object(i.jsx)("h1",{className:"font-bold mb-3 text-xl mt-6",children:"The sequences"}),Object(i.jsxs)("p",{children:["This tree displays sequences generated by scientists all over the world. The sequences are sourced from GenBank (i.e. the INSDC databases), the China National Center for Bioinformation, and from COG-UK. They are collated"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"http://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2//",children:"here"})," ","by Angie Hinrichs and others at UCSC."]}),Object(i.jsx)("p",{children:"This website would be empty but for the hard work of all these researchers, and their openness in submitting their work to public databases, and we gratefully acknowledge this."}),Object(i.jsx)("h1",{className:"font-bold mb-3 text-xl mt-6",children:"The tree"}),Object(i.jsxs)("p",{children:["The tree shown here is produced by the UCSC team, and made available"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"http://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2//",children:"here"}),". It is derived from releases of Rob Lanfear's"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://github.com/roblanf/sarscov2phylo",children:"sarscov2phylo"}),", pruned to include only public sequences aggregated from GenBank, COG-UK, and the China National Center for Bioinformation. The tree has been re-rooted to Wuhan/Hu-1 (GenBank MN908947.3, RefSeq NC_045512.2), and nodes with no associated mutations have been collapsed. Sequences released after the final sarscov2phylo release (Nov. 13, 2020) have been added to the tree using"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://github.com/yatisht/usher",children:"UShER"}),", developed by Yatish Turakhia. UShER development and tree inference was done in Russ Corbett-Detig and David Haussler's groups at the UCSC Genomics Institute. Tree-building work is carried out by the UCSC team and described here only for completeness. They provide metadata with the sequences, including lineages called by"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://www.pango.network/",children:"Pangolin"}),"."]}),Object(i.jsx)("h1",{className:"font-bold mb-3 text-xl mt-4",children:"Citations"}),Object(i.jsxs)("div",{className:"text-sm",children:[Object(i.jsx)("p",{children:"Lanfear, Rob (2020). A global phylogeny of SARS-CoV-2 sequences from GISAID. Zenodo DOI: 10.5281/zenodo.3958883"}),Object(i.jsx)("p",{children:"Turakhia et al. (2021). Ultrafast Sample Placement on Existing Trees (UShER) Empowers Real-Time Phylogenetics for the SARS-CoV-2 Pandemic."}),Object(i.jsx)("p",{children:"Rambaut et al. (2020). A dynamic nomenclature proposal for SARS-CoV-2 lineages to assist genomic epidemiology."}),Object(i.jsxs)("p",{children:["McBroome et al. (2020). ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://www.biorxiv.org/content/10.1101/2021.04.03.438321v2/",children:"A daily-updated database and tools for comprehensive SARS-CoV-2 mutation-annotated trees"})]})]}),Object(i.jsx)("h1",{className:"font-bold mb-3 text-xl mt-4",children:"Related tools"}),Object(i.jsx)("div",{className:"text-sm",children:Object(i.jsxs)("p",{children:["You can also explore large phylogenies using"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://pando.tools",children:"Pando"}),","," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://microreact.org/",children:"Microreact"}),", and sampled phylogenies on"," ",Object(i.jsx)("a",{className:"text-blue-700 underline",href:"https://nextstrain.org/",children:"NextStrain"}),"."]})})]})," "]})]})},h=a(12),u=a(25),b=a(19),j=a(3),m=function(e){return"?"+Object.keys(e).filter((function(t){return void 0!==e[t]})).map((function(t){return"".concat(t,"=").concat(e[t])})).join("&")},f=function(e){for(var t={},a=0,n=Object.entries(e);a<n.length;a++){var s=Object(l.a)(n[a],2),r=s[0],o=s[1];void 0!==o&&(t[r]=encodeURIComponent(o))}return t},x=function(e){var t=Object(j.d)(),a=Object(j.e)(),s=a.pathname,r=a.search,o=Object(n.useMemo)((function(){return function(e){var t={};return new URLSearchParams(e).forEach((function(e,a){t[a]=e})),t}(r)}),[r]),i=Object(n.useRef)({decodedSearch:o,pathname:s});Object(n.useEffect)((function(){i.current={decodedSearch:o,pathname:s}}),[o,s]);var d=Object(n.useCallback)((function(e){var a=arguments.length>1&&void 0!==arguments[1]?arguments[1]:"push",n=i.current,s=n.pathname,r=n.decodedSearch;t[a](s+m(f(Object(c.a)(Object(c.a)({},r),e))))}),[t]);return[Object(n.useMemo)((function(){return Object.assign({},e,function(e){for(var t={},a=0,n=Object.entries(e);a<n.length;a++){var s=Object(l.a)(n[a],2),r=s[0],o=s[1];void 0!==o&&(t[r]=o)}return t}(o))}),[o,e]),d]},p=s.a.lazy((function(){return Promise.all([a.e(2),a.e(4)]).then(a.bind(null,280))})),g=s.a.lazy((function(){return a.e(5).then(a.bind(null,271))}));var O=function(){function e(e){var t=new FileReader;t.onload=function(){N(t.result)},t.readAsArrayBuffer(e)}var t=x({search:JSON.stringify([{id:.123,category:"lineage",value:"",enabled:!0,aa_final:"any",min_tips:1,aa_gene:"S",search_for_ids:""}]),colourBy:JSON.stringify({variable:"lineage",gene:"S",colourLines:!1,residue:"681"})}),a=Object(l.a)(t,2),s=a[0],r=a[1],o=Object(n.useState)(!1),j=Object(l.a)(o,2),m=j[0],f=j[1],O=Object(n.useState)(null),v=Object(l.a)(O,2),y=v[0],N=v[1],w=Object(n.useState)(!1),S=Object(l.a)(w,2),C=S[0],T=S[1],R=Object(n.useState)(""),k=Object(l.a)(R,2),U=k[0],A=k[1];return Object(i.jsxs)(h.a,{children:[Object(i.jsx)(d,{enabled:C,setEnabled:T}),Object(i.jsxs)("div",{className:"h-screen w-screen",onDrop:function(t){if(console.log("File(s) dropped"),t.preventDefault(),t.dataTransfer.items)for(var a=0;a<t.dataTransfer.items.length;a++){if("file"===t.dataTransfer.items[a].kind)e(t.dataTransfer.items[a].getAsFile()),f(!1)}else e(t.dataTransfer.files[0]),f(!1)},onDragOver:function(e){console.log("File(s) in drop zone"),f(!0),e.preventDefault()},onDragLeave:function(e){f(!1),e.preventDefault()},children:[m&&Object(i.jsx)("div",{children:"Drop file to upload"}),Object(i.jsx)("div",{className:"from-gray-500 to-gray-600 bg-gradient-to-bl h-15 shadow-md z-20",children:Object(i.jsxs)("div",{className:"flex justify-between",children:[Object(i.jsxs)("h1",{className:"text-xl p-4  pb-5 text-white ",children:[Object(i.jsx)(u.a,{className:"inline-block h-8 w-8 pr-2 "}),Object(i.jsx)("span",{className:"font-bold",children:"Taxodium"})]}),Object(i.jsx)("div",{className:"inline-block p-4 pr-0",children:Object(i.jsxs)("button",{onClick:function(){return T(!0)},className:"mr-5 text-white font-bold hover:underline",children:[Object(i.jsx)(b.b,{className:"inline-block h-7 w-8"})," About / Acknowledgements"]})})]})}),Object(i.jsx)(n.Suspense,{fallback:Object(i.jsx)("div",{children:"Loading..."}),children:y||s.protoUrl?Object(i.jsx)(p,{uploadedData:y,query:s,setQuery:r}):Object(i.jsxs)("div",{className:"m-10",children:[Object(i.jsx)("p",{className:"text-lg text-gray-700 mb-5",children:"Welcome to Taxodium, a tool for exploring large trees"}),Object(i.jsxs)("div",{className:"grid grid-cols-2  divide-x divide-gray-300",children:[Object(i.jsxs)("div",{className:"p-5",children:[Object(i.jsx)("h3",{className:"text-md text-gray-700 font-semibold mb-2",children:"Upload a Taxodium protobuf file"}),Object(i.jsx)(g,{readFile:e,protoUrl:s.protoUrl})]}),Object(i.jsxs)("div",{className:"p-5",children:[Object(i.jsx)("h3",{className:"text-md text-gray-700 font-semibold mb-2",children:"Provide a URL to a Taxodium protobuf file"}),"URL: ",Object(i.jsx)("input",{type:"text",className:"border-gray-300 p-1 w-60 border",value:U,onChange:function(e){return A(e.target.value)}}),Object(i.jsx)("br",{}),Object(i.jsx)("button",{className:"  bg-gray-100 text-sm mx-auto p-1 rounded border-gray-300 border  text-gray-700 ml-8 h-8 mt-5",onClick:function(){return r(Object(c.a)(Object(c.a)({},s),{},{protoUrl:U.replace("http://","https://")}))},children:"Load"})]})]}),Object(i.jsxs)("p",{className:"text-md text-gray-700 font-semibold mb-2",children:["or ",Object(i.jsx)("a",{className:"text-blue-500",href:"/?protoUrl=/https://hgwdev.gi.ucsc.edu/~angie/UShER_SARS-CoV-2/public-latest.all.masked.taxodium.pb.gz",children:"load the public SARS-CoV-2 tree"}),"."]})," "]})})]})]})},v=function(e){e&&e instanceof Function&&a.e(6).then(a.bind(null,272)).then((function(t){var a=t.getCLS,n=t.getFID,s=t.getFCP,r=t.getLCP,o=t.getTTFB;a(e),n(e),s(e),r(e),o(e)}))};o.a.render(Object(i.jsx)(h.a,{children:Object(i.jsx)(O,{})}),document.getElementById("root")),v()}},[[38,1,3]]]);
//# sourceMappingURL=main.ef3de314.chunk.js.map